connutils.go

```diff
@@ -45,8 +45,59 @@ type connConfig struct {
 	database           string
 	connectTimeout     time.Duration
 	waitUntilAvailable time.Duration
+	tlsCAData          []byte
+	tlsSecurity        string
 	serverSettings     map[string]string
-	tlsConfig          *tls.Config
+}
+
+func (c *connConfig) tlsConfig() (*tls.Config, error) {
+	var roots *x509.CertPool
+	if len(c.tlsCAData) != 0 {
+		roots = x509.NewCertPool()
+		if ok := roots.AppendCertsFromPEM(c.tlsCAData); !ok {
+			return nil, errors.New("invalid certificate data")
+		}
+	} else {
+		var err error
+		roots, err = getSystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	tlsConfig := &tls.Config{
+		RootCAs:    roots,
+		NextProtos: []string{"edgedb-binary"},
+	}
+
+	security, err := getEnvVarSetting(
+		"EDGEDB_CLIENT_SECURITY", "default", "default", "insecure_dev_mode")
```

Comment:
Add `strict`, which is the default.

Reply:
I'm confused. This is the replacement for `EDGEDB_INSECURE_DEV_MODE`. Is the default really `strict` TLS?

Reply:
Shouldn't the default be to have no effect? Then whatever `tls_security` workouts out to be (who's default is `strict`) will be the end result.

Reply:
Hm, you're right. The default is the same as `EDGEDB_CLIENT_TLS_SECURITY`, but `strict` should still be an option.